runtime/src/append_vec.rs

```diff
@@ -187,17 +199,39 @@ impl AppendVec {
 
         let map = unsafe { MmapMut::map_mut(&data)? };
         self.map = map;
+
+        if !self.sanitize_layout_and_length() {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::Other,
+                "incorrect layout/length",
+            ));
+        }
+
         Ok(())
     }
 
+    fn sanitize_layout_and_length(&self) -> bool {
+        let mut offset = 0;
+
+        while let Some((account, next_offset)) = self.get_account(offset) {
```

Comment:
This discards allocated accounts immediately after check at each loop iteration.

This code should not reuse `AppendVec.accounts()` method as the current form or extend it to be reused here because it would allow attackers could accumulate some measurable amount of memory needlessly.

Reply:
I think add a similar comment to the function here.

Reply:
yeah, I just noticed this should be a source comment, too! dadb2e0e05c9f9f2663302c96ee1bb42ae682a8d